DigitalLearningSolutions.Data/Services/DelegateApprovalsService.cs

```diff
@@ -99,6 +100,37 @@ private void SendDelegateApprovalEmails(params DelegateUser[] delegateUsers)
 
             emailService.SendEmails(approvalEmails);
         }
+        public void RejectDelegate(int delegateId, int centreId)
+        {
+            var delegateUser = userDataService.GetDelegateUserById(delegateId);
+
+            if (delegateUser == null || delegateUser.CentreId != centreId)
+            {
+                throw new UserAccountNotFoundException($"Delegate user id {delegateId} not found at centre id {centreId}.");
+            }
+            if (delegateUser.Approved)
+            {
+                logger.LogWarning($"Delegate user id {delegateId} cannot be rejected as they are already approved.");

```

Comment:
I wonder if in addition to logging we want to take the user to an "Unexpected error" page in this case, just to demonstrate that their Rejection action has not been successful..